doc/rtd/topics/modules.rst

```diff
@@ -3,6 +3,10 @@
 *******
 Modules
 *******
+
+This page documents each of the available cloud config modules available in
+cloud-init. Any new modules should use underscores in any new config options.
```

Comment:
Is this user-facing documentation the appropriate place to record this developer-facing decision? (Would a section in HACKING be more apropos?)

Reply:
I originally went to the HACKING document as well. In its current state it is more about how to get started on cloud-init. However, there are other bugs, like [LP: #1838092](https://bugs.launchpad.net/cloud-init/+bug/1838092), that would be a candidate for putting something in HACKING.

What do you think about starting a "Design" Section in there?

Reply:
Yeah, that sounds like a great idea!